src/Features/Core/Portable/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs

```diff
@@ -23,9 +23,10 @@ internal abstract class AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer
         // The NotConfigurable custom tag ensures that user can't turn this diagnostic into a warning / error via
         // ruleset editor or solution explorer. Setting messageFormat to empty string ensures that we won't display
         // this diagnostic in the preview pane header.
+        // Setting category to "Compiler" ensures we always run this analyzer even when user has turned off analyzer execution for the project.
         private static readonly DiagnosticDescriptor s_fixableIdDescriptor =
             new DiagnosticDescriptor(DiagnosticFixableId,
-                                     title: "", messageFormat: "", category: "",
+                                     title: "", messageFormat: "", category: DiagnosticCategory.Compiler,
```

Comment:
❗️ This is not a compiler analyzer. If the user disables live analysis, this analyzer must not run.

Reply:
IMO, majority of users see this a built-in VS-only refactoring, not an analyzer and this is very widely used. Personally, I would like to keep this running even when analyzers are disabled. However, I agree that this does not need to be done as part of this PR. I will revert this change and we can discuss this separately or just wait for user feedback on this.